### PR TITLE
chore: migrate codebase to typescript

### PR DESCRIPTION
**Pull request by DominicKramer**
BREAKING CHANGE:  As a result of the migration, the way in which this package is required and used has changed slightly.

That is, the package is required and a `LoggingWinston` object is created as follows on Node 6+.  In particular, note the `{}` braces used in the require statement and the use of `new` which is now mandatory.

```javascript
// Imports the Google Cloud client library for Winston
const {LoggingWinston} = require('@google-cloud/logging-winston');

// Creates a Winston Stackdriver Logging client
const loggingWinston = new LoggingWinston();
```

On Node 4, the  require syntax is slightly different, but the use of `new` is still required.

```javascript
// Imports the Google Cloud client library for Winston
const LoggingWinston = require('@google-cloud/logging-winston').LoggingWinston;

// Creates a Winston Stackdriver Logging client
const loggingWinston = new LoggingWinston();
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -223,7 +223,7 @@ jobs:
                 -k "${SYSTEM_TESTS_ENCRYPTION_KEY}"
       - run:
           name: Install modules and dependencies.
-          command: npm install
+          command: npm install --unsafe-perm
```
Comment (ofrobots) [inline .circleci/config.yml]: Did this sneak back in?

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
     "node": ">=4.0.0"
   },
   "repository": "googleapis/nodejs-logging-winston",
-  "main": "./src/index.js",
+  "main": "./build/src/index.js",
   "files": [
     "src",
```
Comment (ofrobots) [inline package.json]: Did you miss the changes to this section?

File: .circleci/config.yml (hunk 1 above)
Comment (DominicKramer) [reply .circleci/config.yml]: Fixed

File: package.json (hunk 2 above)
Comment (DominicKramer) [reply package.json]: I've added a `types` attribute and updated the `files` section.

**Pull request merged**